# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/bindings/Exceptions.cpp
# Commit: e4783542b7c8
# Full Hash: e4783542b7c854ae4aa5cf60ac27e5e1ff19d56a
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-12-07 21:55:34
# Regressor Bug: 1588839
# File Overlap Count: 1
# Description:
#   Bug 1588839 - Part 1 - Add helper to allow structured cloning Error/Exception objects. r=bzbarsky
#   
#   In order to be able to reasonably debug error results from things like
#   JSWindowActor.sendQuery, we need to be able to clone errors across process
#   boundaries, so that they can be propagated to the caller that initiated a
# ==============================================================================

diff -r df371f905d3a -r e4783542b7c8 dom/bindings/Exceptions.cpp
--- a/dom/bindings/Exceptions.cpp	Sat Dec 07 17:45:53 2019 +0000
+++ b/dom/bindings/Exceptions.cpp	Sat Dec 07 18:59:14 2019 +0000
@@ -742,12 +742,15 @@
     return nullptr;
   }
 
-  if (!stack) {
-    return nullptr;
+  return CreateStack(aCx, stack);
+}
+
+already_AddRefed<nsIStackFrame> CreateStack(JSContext* aCx,
+                                            JS::Handle<JSObject*> aStack) {
+  if (aStack) {
+    return MakeAndAddRef<JSStackFrame>(aStack);
   }
-
-  nsCOMPtr<nsIStackFrame> frame = new JSStackFrame(stack);
-  return frame.forget();
+  return nullptr;
 }
 
 }  // namespace exceptions